atlasdb-client-protobufs/src/main/proto/TableMetadataPersistence.proto

```diff
@@ -108,6 +108,7 @@ enum TableConflictHandler {
     SERIALIZABLE_CELL = 6;
     SERIALIZABLE_INDEX = 7;
     SERIALIZABLE_LOCK_LEVEL_MIGRATION = 8;
+    RETRY_ON_WRITE_WRITE_MIGRATION = 9;
```

Comment:
The main danger here is that another node reading `RETRY_ON_WRITE_WRITE_MIGRATION` that doesn't know this enum value exists will read `IGNORE_ALL`, which would be bad.

> Notably, unrecognized enum values are discarded when the message is deserialized, which makes the field's `has..` accessor return false and its getter return the first value listed in the enum definition, or the default value if one is specified.

This is fine: we don't have a way of retrofitting changes to old versions though. This means that users of this level (which I understand to largely be atlasdb-proxy) need to follow a careful schema upgrade process. Supposing the current product version is P1 and schema version is S1, you need to

- release (P2, S2) with this version of AtlasDB, where rollback from S2 to S1 is not allowed.
- release (P3, S3) with the table set to RETRY_ON_WRITE_WRITE_MIGRATION, where rollback from S3 to S2 is allowed.
- release (P4, S4) with the table set to RETRY_ON_WRITE_WRITE, where rollback from S4 to S3 is allowed.

This is complex and can go wrong, so we need docs.

I also think we should make the protobuf deserialization code more robust to this (we just hit the getter directly, while if we see it doesn't exist we should explode) but we can't retrofit old code anyway so the above process is required.